### ISSUE TITLE
Corrupted sprite in Burning Spear demo

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#692]
[Originally posted](https://github.com/vAmigaWeb/vAmigaWeb/issues/77) by @Vweber73:

> See this interesting post from Toni, Burning Spear demo is supposed to have a corrupt sprite in the middle of the screen only on OCS Amiga. But vAmigaWeb shows this corrupt sprite even when ECS Agnus is selected !

Related EAB article: http://eab.abime.net/showthread.php?t=11295&page=2

vAmiga: 
<img width="1145" alt="Bildschirmfoto 2022-05-01 um 13 11 19" src="https://user-images.githubusercontent.com/12561945/166143575-1b93603c-ebb5-4270-aec6-f7f5d5e89c3a.png">

**Comment by dirkwhoffmann** [comment]
Fixed. ECS Agnus treats bits 5 and 6 in SPRxCTL as additional position bits.

<img width="1050" alt="Bildschirmfoto 2022-05-19 um 14 47 47" src="https://user-images.githubusercontent.com/12561945/169299677-f7e0491d-b8a7-4cb5-acff-c544615148f2.png">

Seems like an undocumented feature. The HRM doesn't mentions this, not even in the third edition:
> Bits 6-3 are reserved for future use (make zero).

**Comment by Vweber73** [comment]
Wonderful ! Yes, not sure how Toni got this info, indeed the HRM is not crystal clear about this...
Is this already in vAmigaWeb or do I have to wait for it to be rolled out ?
Many thanks and cheers

**Comment by mithrendal** [comment]
soon 😀 very soon ... be well prepared for complete total sprite awesomeness 😂

**Comment by Vweber73** [comment]
Btw, while we are at the chipset, any plan to support the full ECS (Super Denise) ? 
Any plan to support AGA in the long run ?
Many thanks and cheers

**Comment by mras0** [comment]
> Seems like an undocumented feature. The HRM doesn't mentions this, not even in the third edition

I can't find them either right now in an official reference, but FYI they're in the widely circulated "aga.guide" (distributed with e.g. [Asm-one](http://aminet.net/package/dev/asm/ASM-One)).

```
         +-------+----------+------------------------------------------------+
         | BIT#  | SYM      | FUNCTION                                       |
         +-------+----------+------------------------------------------------+
         | 15-08 | EV7-EV0  | End (stop) vert. value. Low 8 bits             |
         | 07    | ATT      | Sprite attach control bit (odd sprites only)   |
         | 06    | SV9      | Start vert value 10th bit.                     |
         | 05    | EV9      | End (stop) vert. value 10th bit                |
         | 04    | SH1=0    | Start horiz. value, 70nS increment             |
         | 03    | SH0=0    | Start horiz. value 35nS increment              |
         | 02    | SV8      | Start vert. value 9th bit                      |
         | 01    | EV8      | End (stop) vert. value 9th bit                 |
         | 00    | SH2      | Start horiz.value,140nS increment              |
         +-------+----------+------------------------------------------------+
```

> Yes, not sure how Toni got this info, indeed the HRM is not crystal clear about this...

Lots and lots of testing. If you're interested in that kind of stuff check out the [undocumented hw stuff thread](http://eab.abime.net/showthread.php?t=19676) on EAB. Dirk does that as well [vAmigaTS](https://github.com/dirkwhoffmann/vAmigaTS). This issue is pretty simple as these things go. I know that for some of the later stuff it took many weeks with testing and looking at logic analyzer traces.

I won't comment on any timelines/plans for vAmiga, but all (interesting) AGA software requires 68020+ so that would have to be done first. True ECS support (i.e. actually supporting programmed resolutions and not just hacks for the common ones) I also think was a very recent addition to WinUAE that took an insane effort, so probably unlikely to occur any time soon.

**Comment by Vweber73** [comment]
Very interesting, thanks !

Yes, as I said for OCS/ECS software 68000 instructions are enough, overclocking will bring all the benefits you could except from a higher processor (but let's be careful to not be faster than the blitter, or to wait for it, this is a source of incompatibilities as I understand it). AGA is another story, yes it often requires 68020+...

So, Super Denise was very hard to implement in WinUAE, and that was only done recently ? Didn't know that i was so complex. Anyway Super Denise is useless for games and demos which is the main usage...

Cheers

**Comment by mras0** [comment]
Yes, as you mention no games/demos use the advanced ECS features, so what I think happened is that (Win)UAE just supported what's necessary for what was configurable from WB. The difficult part was supporting all the options that are actually possible (necessitating a rewrite of the display logic as far as I understand). Since no existing software seemed to use this flexibility, this was (AFAIK) only done to make emulation more accurate.

The only further ECS stuff I see an immediate benefit to is supporting a flicker-free 640x400/512 mode for productivity software.

**Comment by Vweber73** [comment]
Agreed !
Cheers

**Comment by mithrendal** [comment]
>Is this already in vAmigaWeb or do I have to wait for it to be rolled out ?

merged and pushed out todays latest commits of vAmiga into vAmigaWeb ... i.e. support for overclocking and ECS sprite corruption fix ...

![image](https://user-images.githubusercontent.com/17108995/169400114-0203449f-b0d3-4d2a-9f97-dcccdbc83db0.png)

**Comment by dirkwhoffmann** [comment]
Fixed in v2.1b1.
